### PR TITLE
Portability Fixes

### PR DESCRIPTION
**Pull request by Mic92**
### Description/Motivation/Screenshots ###

Fixes portability for macOS/FreeBSD/NixOS

### How Has This Been Tested? ###

| Architecture |          Yes/No          | Comments                                  |
| ------------ | :----------------------: | ----------------------------------------- |
| x86-32       | :heavy_multiplication_x: | Replace with :heavy_check_mark: if tested |
| x86-64       | :heavy_check_mark: |                                           |
| ARM          | :heavy_multiplication_x: |                                           |
| AARCH64      | :heavy_multiplication_x: |                                           |
| MIPS         | :heavy_multiplication_x: |                                           |
| POWERPC      | :heavy_multiplication_x: |                                           |
| SPARC        | :heavy_multiplication_x: |                                           |
| RISC-V       | :heavy_multiplication_x: |                                           |
| `make tests` | :heavy_multiplication_x: |                                           |

### Checklist ###

- [x] My PR was done against the `dev` branch, not `master`.
- [] My code follows the code style of this project.
- [] My change includes a change to the documentation, if required.
- [] My change adds tests as appropriate.
- [x] I have read and agree to the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by Grazfather** [comment]
LGTM

File: gef.py [hunk 1]
```diff
@@ -5485,7 +5485,7 @@ def set_fs(uc, addr):    return set_msr(uc, FSMSR, addr)
 """
 
 
-        content = """#!/usr/bin/python -i
+        content = """#!/usr/bin/bin/env -S python -i
```
Comment (hugsy) [inline gef.py:5488]: `env -S` doesn't exist on (at least) Ubuntu 18.04 LTS which is the most used distro for GEF. Please adjust.

Comment (hugsy) [inline gef.py:5488]: Also `/usr/bin/bin/env` is unlikely to be a valid path. Please fix.

File: gef.py [hunk 2]
```diff
@@ -6886,7 +6886,7 @@ class ProcessListingCommand(GenericCommand):
 
     def __init__(self):
         super(ProcessListingCommand, self).__init__(complete=gdb.COMPLETE_LOCATION)
-        self.add_setting("ps_command", "/bin/ps auxww", "`ps` command to get process information")
+        self.add_setting("ps_command", "ps auxww", "`ps` command to get process information")
```
Comment (hugsy) [inline gef.py:6889]: Use the `which()` function to return the absolute path of `ps`. This function has the advantage of raising a `FileNotFoundError` exception in the (unlikely) `ps` is not found in the PATH.

File: gef.py (hunk 1 above)
Comment (Mic92) [reply gef.py:5488]: Looks like coreutils is too old. I could switch to using the `which()` function to check which python to put in the shebang.

Comment (hugsy) [reply gef.py:5488]: Yep, `which()` would be an acceptable solution.

File: gef.py (hunk 2 above)
Comment (Mic92) [reply gef.py:6889]: Where should be called?

File: gef.py (hunk 1 above)
Comment (Mic92) [reply gef.py]: I use `which()` now.

Comment (Mic92) [reply gef.py]: done.

File: gef.py [hunk 3]
```diff
@@ -5484,8 +5484,13 @@ def set_fs(uc, addr):    return set_msr(uc, FSMSR, addr)
     set_gs(emu, SEGMENT_GS_ADDR)
 """
 
+        try:
```
Comment (Mic92) [inline gef.py:5487]: This part is untested. I currently don't have unicorn installed.

File: gef.py (hunk 2 above)
Comment (hugsy) [reply gef.py:6894]: Something along the lines of :
```python
    def __init__(self):
        [...]
        ps = which("ps")
        self.add_setting("ps_command", "{:s} auxww".format(ps), "`ps` command to get process information")
```
would be better. If `ps` is not found in the PATH, the line `ps = which("ps")` will raise an exception that the user will know about and fix.

File: gef.py (hunk 3 above)
Comment (hugsy) [reply gef.py:5487]: Actually the `try`/`except` is not even necessary: if `python` is not in the PATH, we want a noisy error for the user.

Comment (Mic92) [reply gef.py]: Fixed now.

File: gef.py (hunk 2 above)
Comment (Mic92) [reply gef.py]: done.

**Comment by Mic92** [comment]
Now I was also able to test the unicorn-emulate function.

**Comment by hugsy** [comment]
That's great. 

Last step now, for trackability issue, please [`git-squash`](https://www.devroom.io/2011/07/05/git-squash-your-latests-commits-into-one/) your commits into a single one (also adding the issue number in the commit). It'll then be ready for merge.

Thanks

**Comment by Mic92** [comment]
Done.

**Comment by hugsy** [comment]
Thank you for the contribution.

**Pull request merged**